packages/wallets/src/components/WalletsCarouselContent/WalletsCarouselContent.tsx

```diff
@@ -30,6 +30,65 @@ const WalletsCarouselContent: React.FC<TProps> = ({ onWalletSettled }) => {
     // for the embla "on select" callback
     // to avoid unbinding / cleaning etc, just let it use up-to-date list
     const walletsAccountsListRef = useRef(walletAccountsList);
+    const tweenNodes = useRef<HTMLElement[]>([]);
+    const tweenFactor = useRef(0);
+
+    const numberWithinRange = (number: number, min: number, max: number): number =>
+        Math.min(Math.max(number, min), max);
+
+    // scale based on the width difference between active and inactive wallets
+    const tweenFactorBase = 1 - 24 / 28.8;
+
+    const setTweenNodes = useCallback((emblaApi: EmblaCarouselType) => {
+        tweenNodes.current = emblaApi.slideNodes().map(slideNode => {
+            return slideNode.querySelector('.wallets-card__container') as HTMLElement;
+        });
+    }, []);
+
+    const setTweenFactor = useCallback(
+        (emblaApi: EmblaCarouselType) => {
+            tweenFactor.current = tweenFactorBase * emblaApi.scrollSnapList().length;
+        },
+        [tweenFactorBase]
+    );
+
+    const tweenScale = useCallback((emblaApi: EmblaCarouselType, eventName?: EmblaEventType) => {
+        const engine = emblaApi.internalEngine();
+        const scrollProgress = emblaApi.scrollProgress();
+        const slidesInView = emblaApi.slidesInView();
+        const isScrollEvent = eventName === 'scroll';
+
+        emblaApi.scrollSnapList().forEach((scrollSnap, snapIndex) => {
+            let diffToTarget = scrollSnap - scrollProgress;
+            const slidesInSnap = engine.slideRegistry[snapIndex];
+
+            slidesInSnap.forEach(slideIndex => {
+                if (isScrollEvent && !slidesInView.includes(slideIndex)) return;
+
+                if (engine.options.loop) {
+                    engine.slideLooper.loopPoints.forEach(loopItem => {
+                        const target = loopItem.target();
+
+                        if (slideIndex === loopItem.index && target !== 0) {
+                            const sign = Math.sign(target);
+
+                            if (sign === -1) {
+                                diffToTarget = scrollSnap - (1 + scrollProgress);
+                            }
+                            if (sign === 1) {
+                                diffToTarget = scrollSnap + (1 - scrollProgress);
+                            }
+                        }
+                    });
+                }
+
+                const tweenValue = 1 - Math.abs(diffToTarget * tweenFactor.current);
+                const scale = numberWithinRange(tweenValue, 0, 1).toString();
+                const tweenNode = tweenNodes.current[slideIndex];
+                tweenNode.style.transform = `scale(${scale})`;
+            });
+        });
```

Comment:
Can we add some comments or rename it with something meaningful so the others can understand fully what the function is and what does it do.

Reply:
Added some comments and renamed the functions to make it more understandable. lmk if anything else needs more clarification 🙏

Reply:
Here's a reference for the embla scaling function from the embla documentation: https://codesandbox.io/p/sandbox/embla-carousel-scale-react-r5rp4q?file=%2Fsrc%2Fjs%2FEmblaCarousel.tsx